auth/src/main/java/org/iqu/auth/passwordreset/EmailSender.java

```diff
@@ -0,0 +1,62 @@
+package org.iqu.auth.passwordreset;
+
+import static org.iqu.auth.passwordreset.PropertiesConstants.SMTP_PASSWORD;
+import static org.iqu.auth.passwordreset.PropertiesConstants.SMTP_USER;
+import java.util.Properties;
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import org.apache.log4j.Logger;
+import org.iqu.auth.exception.ConfigurationException;
+import org.iqu.auth.token.TokenGenerator;
+
+/**
+ * Send email with reset token.
+ * 
+ * @author Mitroi Stefan-Daniel
+ *
+ */
+public class EmailSender {
+
+	private Authenticator auth;
+	private Properties props;
+	private PropertiesLoader propsLoader;
+	private TokenGenerator tokenGenerator;
+	private String messageText;
+	private String messageSubject;
+	private static final Logger LOGGER = Logger.getLogger(EmailSender.class);
+
+	public EmailSender() {
+
+		propsLoader = new PropertiesLoader();
+		tokenGenerator = new TokenGenerator();
+	}
+
+	public String sendMail(String userName, String toEmail) {
+
+		try {
+			props = propsLoader.loadProperties();
+		} catch (ConfigurationException e) {
+			LOGGER.error("can't load properties", e);
+		}
+		auth = new SMTPAuthenticator(props.getProperty(SMTP_USER), props.getProperty(SMTP_PASSWORD));
```

Comment:
In case an exception is previously thrown at the loading of properties that it will be catched and logged, but then your props object will be null and another error such as NullPointerException will be thrown for : props.getProperty(SMTP_USER)